Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/TicketService.hs

```diff
@@ -594,3 +632,222 @@ postTicketBookingsUpdateSeats _ TicketBookingUpdateSeatsReq {..} = do
+                ttl = Nothing,
+                sound = notificationSound
+              }
+          title = "Ticket Cancelled"
+          body =
+            unwords
+              [ "Sorry, Ticket booking has been cancelled And Refunded",
+                "Check the app for details"
+              ]
+      TNotification.notifyPerson person.merchantId person.merchantOperatingCityId notificationData
+
+postTicketBookingCancel :: (Maybe (Kernel.Types.Id.Id Domain.Types.Person.Person), Kernel.Types.Id.Id Domain.Types.Merchant.Merchant) -> API.Types.UI.TicketService.TicketBookingCancelReq -> Environment.Flow API.Types.UI.TicketService.TicketBookingCancelResp
+postTicketBookingCancel _ req = do
+  ticketBooking <- QTB.findByShortId req.ticketBookingShortId >>= fromMaybeM (TicketBookingNotFound req.ticketBookingShortId.getShortId)
+  validateCancellationRequest ticketBooking
+  refundsAmountRequested <- processCancelTicketBookingServices ticketBooking req.ticketBookingServices
+  let totalRefundAmount = sum refundsAmountRequested
+  person <- runInReplica $ QP.findById ticketBooking.personId >>= fromMaybeM (PersonNotFound ticketBooking.personId.getId)
+  initiatedRefundResponse <- intializeRefundProcess ticketBooking totalRefundAmount person.merchantId person.merchantOperatingCityId
+  case initiatedRefundResponse of
+    Left _ -> throwError $ InternalError "Failed to create refund"
+    Right rsp -> pure $ API.Types.UI.TicketService.TicketBookingCancelResp rsp.refundAmount rsp.message
+  where
+    validateCancellationRequest tb = unless (tb.status == DTTB.Booked) $ throwError $ TicketBookingNotConfirmed tb.shortId.getShortId
+    processCancelTicketBookingServices ticketBooking = mapM (cancelTicketBookingService ticketBooking)
+
+cancelTicketBookingService :: DTTB.TicketBooking -> API.Types.UI.TicketService.TicketBookingServiceCancelReq -> Environment.Flow HighPrecMoney
+cancelTicketBookingService ticketBooking tbServiceCancelReq = do
+  ticketBookingService <- QTBS.findByShortId tbServiceCancelReq.shortId >>= fromMaybeM (TicketBookingServiceNotFound tbServiceCancelReq.shortId.getShortId)
+  ticketService <- QTS.findById ticketBookingService.ticketServiceId >>= fromMaybeM (TicketServiceNotFound ticketBookingService.ticketServiceId.getId)
+  validateCancellationRequest ticketBookingService.status ticketService.allowCancellation
+  cancelledSeatsInServiceCategory <- processCancelServiceCategories ticketBooking.visitDate tbServiceCancelReq.serviceCategory
+  let (noOfSeatsToCancel, amountToRefund) = totalSeatsCancelledAndRefundAmount cancelledSeatsInServiceCategory
+
+  currentTime <- getCurrentTime
+  let cancellationPercentage = calculateCancellationCharges ticketBookingService ticketService currentTime ticketBooking.visitDate
+      refundAmount = case cancellationPercentage of
+        Just cancelCharge -> case cancelCharge.charge of
+          Domain.Types.TicketService.FlatFee fee -> amountToRefund - HighPrecMoney (fromIntegral fee)
```

Comment:
Is this flatFee not per head and is per cancellation? @sidsethu 
@max-keviv Can you take maximum of 0 and `amountToRefund - HighPrecMoney (fromIntegral fee)` (this can go negative too)

Reply:
yeah we should have this on per head will change this

Reply:
if we want to have flatfee based on the person then cancellation charges will be on people category